server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQExclusiveConsumer.java

```diff
@@ -0,0 +1,174 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.task.eventsourcing.distributed;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableMap;
+import com.rabbitmq.client.CancelCallback;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.DeliverCallback;
+import com.rabbitmq.client.impl.recovery.AutorecoveringConnection;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+import reactor.rabbitmq.AcknowledgableDelivery;
+import reactor.rabbitmq.ConsumeOptions;
+import reactor.rabbitmq.RabbitFluxException;
+import reactor.rabbitmq.ReceiverOptions;
+
+/**
+ * Taken from {@link reactor.rabbitmq.Receiver}
+ * In order to be able to set the `exclusive` parameter to `true`
+ * to the `channel.basicConsume` method.
+ *
+ * @deprecated to remove once the parallel execution of task has been implemented
+ */
+@Deprecated
+public class RabbitMQExclusiveConsumer implements Closeable {
+    private static final Function<Connection, Channel> CHANNEL_CREATION_FUNCTION = new RabbitMQExclusiveConsumer.ChannelCreationFunction();
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQExclusiveConsumer.class);
+
+    private static final boolean NON_LOCAL = true;
+    private static final boolean EXCLUSIVE = true;
+
+    private static class ChannelCreationFunction implements Function<Connection, Channel> {
+
+        @Override
+        public Channel apply(Connection connection) {
+            try {
+                return connection.createChannel();
+            } catch (IOException e) {
+                throw new RabbitFluxException("Error while creating channel", e);
+            }
+        }
+    }
+
+    private Mono<? extends Connection> connectionMono;
+    private final AtomicBoolean hasConnection;
+    private final Scheduler connectionSubscriptionScheduler;
+    private final boolean privateConnectionSubscriptionScheduler;
+
+    public RabbitMQExclusiveConsumer(ReceiverOptions options) {
+        this.privateConnectionSubscriptionScheduler = options.getConnectionSubscriptionScheduler() == null;
+        this.connectionSubscriptionScheduler = options.getConnectionSubscriptionScheduler() == null ?
+            createScheduler("rabbitmq-receiver-connection-subscription") : options.getConnectionSubscriptionScheduler();
+        hasConnection = new AtomicBoolean(false);
```

Comment:
for the sake of consistency, I would write `this.hasConnection`

Reply:
It's the code from the reactor rabbit Receiver class.
The only changes we did, were to set the exclusive parameter to true when calling the basic consume method.
It is deprecated and it's clearly stated in the comments at the class level.

Reply:
yet adding `this` is very fast to do